### ISSUE TITLE
Add documentation or functionality to address breaking default behavior of Microsoft ODBC Driver 18 for SQL Server

### ISSUE THREAD

**Issue by sfirke** [yehoshuadimarsky/bcpandas#126]
The standard configuration with the minimal arguments works with Microsoft ODBC Driver for SQL Server versions <=17 but not with version 18, which has [breaking security changes](https://techcommunity.microsoft.com/t5/sql-server-blog/odbc-driver-18-0-for-sql-server-released/ba-p/3169228).  In my case the connection fails due to a self-signed certificate.  I want to tell bcpandas to trust the server's certificate.

It would be great to have this addressed, either by modifying `SqlCreds` or simply updating the docs to show an example of how to pass values to `SqlCreds` such that it trusts the server certificate.

**Comment by borchero** [comment]
I'll tackle this in the coming weeks as we'll also need this functionality soon.

**Comment by gordthompson** [comment]
You can just specify the extra connection string argument(s) in `odbc_kwargs`:

```python
creds = SqlCreds(
    server="192.168.0.100",
    database="test",
    username="scott",
    password="tiger^5HHH",
    driver_version=18,
    odbc_kwargs={
        "TrustServerCertificate": "yes",
    },
)
```

**Comment by empowerNate** [comment]
@gordthompson I don't believe the driver_version or odbc_kwargs do anything. Looks like there might be only one kwarg that matters. For the bcp util, you need to add a -u flag to trust the cert. This should be an option somewhere and added to the bcp command string.

**Comment by gordthompson** [comment]
@empowerNate - I dunno, I just tried it and it worked for me (on Windows, at least). I'm using bcp version 15.0.4298.1, released 2023-04-07.

The current [bcp docs](https://learn.microsoft.com/en-us/sql/tools/bcp-utility?view=sql-server-ver16) don't mention a `-u` switch, just `-U` for the login_id.

**Comment by gordthompson** [comment]
@empowerNate - My apologies. bcp version 18.2.0001.1 on Linux does indeed have a `-u` switch and it is required, so `odbc_kwargs` itself is not sufficient. (My previous tests on Linux had bcp version 17.10.0001.1)

**Comment by gordthompson** [comment]
Okay, so

```python
    odbc_kwargs={
        "TrustServerCertificate": "yes",
    },
```

is not effective, but

```python
    odbc_kwargs={
        "Encrypt": "no",
    },
```

does seem to avoid the issue for bcp 18 on Linux.

**Comment by empowerNate** [comment]
Ah ok, I recall that as the only kwarg I saw used in the codebase. Seems a little confusing but 🤷

**Comment by gordthompson** [comment]
Until this is fixed properly, here is a workaround for those who need to trust the server certificate without disabling encryption:

```python
creds = SqlCreds(
    "192.168.0.199",
    "test",
    "sa",
    "scott_tiger^5HHH",
    driver_version=18,
    odbc_kwargs={
        "TrustServerCertificate": "yes",
    }
)

script_path = "/tmp/bcp_u.sh"
with open(script_path, "w") as f:
    f.write("bcp $1 $2 $3 $4 $5 $6 $7 $8 $9 $10 $11 $12 $13 $14 $15 $16 $17 $18 $19 -u")
os.chmod(script_path, stat.S_IRWXU)

to_sql(
    df,
    "my_table",
    creds,
    index=False,
    if_exists="replace",
    bcp_path=script_path,
)

**Comment by RatserX** [comment]
Has there been any progress on this? Our apps have been recently migrated to use ODBC Driver 18. We've implemented the workaround, but there are some questions regarding if a proper solution is being worked on.
